tests/base_cluster.py

```diff
@@ -84,20 +85,39 @@ def get_ip_address_dict(self):
         """
         pass
 
-    @abc.abstractmethod
     def stop_host(self, host_name):
-        """Stops a host. Paradoxically, start_host doesn't seem to be required
-        for the product tests to run successfully."""
-        pass
-
-    @abc.abstractmethod
-    def get_down_hostname(self, host_name):
-        """This is part of the magic involved in stopping a host. If you're
-        rolling a new implementation, you should dig more deeply into the
-        existing implementations, figure out how it all works, and update this
-        comment.
-        """
-        pass
+        """Stop a host by editing the presto-admin topology to point
+        to a non-existent host as provided by get_down_hostname() and
+        get_down_ip()"""
+        if host_name not in self.all_hosts():
+            raise Exception('Must specify external hostname to stop_host')
+
+        # Change the topology to something that doesn't exist
+        ips = self.get_ip_address_dict()
+        down_hostname = self.get_down_hostname()
+        self.exec_cmd_on_host(
```

Comment:
Why do we need all three of these sed commands? Doesn't the topology usually just contain the internal host name?